src/include/miopen/solver.hpp

```diff
@@ -2429,8 +2516,15 @@ struct ConvOclBwdWrW2 : ConvTunableSolver<PerformanceConfigConvOclBwdWrw2<N_BATC
 #if defined(__clang__) && defined(CONV_OCL_DIR2D_BWDWRW_2_CPP)
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wweak-template-vtables"
+#pragma clang diagnostic ignored "-Wunneeded-internal-declaration"
 #endif
 
+extern template struct PerformanceConfigConvOclBwdWrw2<1>;
+extern template struct PerformanceConfigConvOclBwdWrw2<2>;
+extern template struct PerformanceConfigConvOclBwdWrw2<4>;
+extern template struct PerformanceConfigConvOclBwdWrw2<8>;
+extern template struct PerformanceConfigConvOclBwdWrw2<16>;
```

Comment:
@cderb Could you please explain why do you need this code to be here?

Reply:
Changes in generic_search required forward declaration of solvers to avoid compiler warnings.